api/base/serializers.py

```diff
@@ -280,6 +299,54 @@ def to_representation(self, data):
             self.child.to_representation(item, envelope=None) for item in data
         ]
 
+    # Overrides ListSerializer which doesn't support multiple update by default
+    def update(self, instance, validated_data):
+        if len(instance) != len(validated_data):
+            raise exceptions.NotFound()
+        id_lookup = self.child.fields['id'].source
+        instance_mapping = {getattr(item, id_lookup): item for item in instance}
+        data_mapping = {item.get('_id', None): item for item in validated_data}
+
+        ret = []
+
+        for resource_id, data in data_mapping.items():
+            resource = instance_mapping.get(resource_id, None)
```

Comment:
I would think this would have been caught in L304.  The instance was formed in get_queryset which returns each resource object in validated_data. If it can't find it, the two lists would be different sizes.